adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -726,6 +726,93 @@ public void onError(Exception exc) {
         return authenticationResult.get();
     }
 
+    /**
+     * This is sync function. It will first look at the cache and automatically
+     * checks for the token expiration. Additionally, if no suitable access
+     * token is found in the cache, but refresh token is available, the function
+     * will use the refresh token automatically. This method will not show UI
+     * for the user. If prompt is needed, the method will return an exception
+     *
+     * @param resource required resource identifier.
+     * @param clientId required client identifier.
+     * @param userId   UserID obtained from
+     *                 {@link AuthenticationResult #getUserInfo()}
+     * @param forceRefresh when true, access token is renewed using broker if available; otherwise, uses local refresh token
+     * @return A {@link Future} object representing the
+     * {@link AuthenticationResult} of the call. It contains Access
+     * Token,the Access Token's expiration time, Refresh token, and
+     * {@link UserInfo}.
+     * @throws AuthenticationException If silent request fails to get the token back.
+     * @throws InterruptedException    If the main thread is interrupted before or during the activity.
+     */
+    public AuthenticationResult acquireTokenSilentSync(String resource, String clientId, String userId, boolean forceRefresh)
+            throws AuthenticationException, InterruptedException {
+
+        final String methodName = ":acquireTokenSilentSync";
+        checkPreRequirements(resource, clientId);
+        checkADFSValidationRequirements(null);
+        final AtomicReference<AuthenticationResult> authenticationResult = new AtomicReference<>();
+        final AtomicReference<Exception> exception = new AtomicReference<>();
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        final String requestId = Telemetry.registerNewRequest();
+        final APIEvent apiEvent = createApiEvent(mContext, clientId, requestId, EventStrings.ACQUIRE_TOKEN_SILENT_SYNC);
```

Comment:
I suspect that this call should get its own `API_ID` -- see [this doc](https://microsoft.sharepoint.com/:w:/r/teams/aad/devex/_layouts/15/Doc.aspx?sourcedoc=%7BAAE38591-8A29-4A50-AE00-B9F66505B6D2%7D&file=Telemetry%20DDS.docx&action=default&mobileredirect=true) (requires corpnet) for a table of API IDs and their meanings. If iOS is adding this functionality, we'll need to collaborate with them to ensure we're using the same values